Raven.Database/Bundles/Replication/Controllers/ReplicationController.cs

```diff
@@ -412,17 +413,21 @@ private void AddSingleConflict(RavenJObject conflict, List<ConflictToResolve> co
             return GetEmptyMessage();
         }
 
-        private void SaveReplicationSource(string src, string lastEtag, int batchSize, string collections = null)
+        private void SaveReplicationSource(string src, string lastEtag, int batchSize, DateTime lastModified, string collections = null)
         {
             var remoteServerInstanceId = Guid.Parse(GetQueryStringValue("dbid"));
 
             var replicationDocKey = Constants.RavenReplicationSourcesBasePath + "/" + remoteServerInstanceId;
 
             var replicationDocument = Database.Documents.Get(replicationDocKey, null);
             var lastAttachmentId = Etag.Empty;
+            DateTime lastModification = lastModified;
             if (replicationDocument != null)
```

Comment:
Tombstones are within the list and everything is ordered by Etag:
https://github.com/ayende/ravendb/blob/v3.5/Raven.Database/Bundles/Replication/Tasks/ReplicationTask.cs#L1284-L1294